### DISCUSSION TITLE
Existing S3 files import to Pocketbase

### DISCUSSION THREAD

**Discussion by jarredwitt** [pocketbase/pocketbase#4468] [Q&A]
Hi all,

I currently have an S3 bucket that has about 40k files in it. Is there a way that I can associate these files that already in S3 with a record that has a file field without having to perform an upload?

Thanks

**Comment by ganigeorgiev** [comment]
Unfortunately I don't think there is an easy way for this.

In the database we store only the base filename (ex. `something_abc123.png`) while expecting the S3 prefix/key to be `:collectionId/:recordId/:filename` (when using the local filesystem this also represents the path to the file under the `pb_data/storage/` directory).

**Comment by jonshipman** [comment]
I'd recommend console commands: https://pocketbase.io/docs/js-console-commands/

You can use console commands and https://pocketbase.io/docs/js-database/ and create a process that A) loops over the files in S3 storage, B) creates a file record, and C) move the file in S3 to the correct spot.

The js-database commands allow you to set file fields as strings in base sqlite commands.

Assuming "file" is the name of the field with the filename and file type in Pocketbase:

Using the S3 API, get your items in your bucket.
Loop over each item.
For Each Item, perform an insert query where the "file" field is the basename for the item (hint: you can use `$security.randomString(15);` to generate an id programatically)
Create a new key in the format `:collectionId/:recordId` and move the S3 file to that location.
